Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -522,13 +526,29 @@ String HTMLAttachmentElement::attachmentTitleForDisplay() const
     if (indexOfLastDot == notFound)
         return title;
 
+    auto filename = StringView(title).left(indexOfLastDot);
+    auto extension = StringView(title).substring(indexOfLastDot);
+
+    if (isWideLayout() && Layout::TextUtil::directionForTextContent(filename) == TextDirection::RTL) {
```

Comment:
using Layout namespace here is a bit of a layering violation (I know this is just a static helper). isn't there a WTF function on this?

Reply:
Ah yep, I see it's using `ubidi_getBaseDirection` from WTF/icu/unicode/ubidi.h, I'll use that as well...